packages/bpk-component-card-list/src/BpkCardList.tsx

```diff
@@ -0,0 +1,165 @@
+
+const getClassName = cssModules(STYLES);
+const MAX_ITEMS = 12; // MAX should be 12 for Desktop Grid and Mobile Stack
+const DEFAULT_ITEMS = 3;
+
+const BpkCardList = (props: BpkCardListProps) => {
+  const {
+    buttonText,
+    cardList,
+    description,
+    initiallyShownCards = DEFAULT_ITEMS,
+    layoutDesktop,
+    layoutMobile,
+    onButtonClick,
+    title,
+  } = props;
+
+  const [visibleCards, setVisibleCards] = useState(
+    cardList.slice(0, initiallyShownCards),
+  );
+  const [collapsed, setCollapsed] = useState(true);
+
+  const showContent = () => {
+    setVisibleCards(cardList.slice(0, MAX_ITEMS));
+
+    onButtonClick?.();
+  };
+
+  const hideContent = () => {
+    setVisibleCards(cardList.slice(0, initiallyShownCards));
+
+    onButtonClick?.();
+  };
+
+  const button = buttonText && (
+    <BpkButtonV2 onClick={onButtonClick}>{buttonText}</BpkButtonV2>
+  );
+
+  return (
+    <div className={getClassName('bpk-card-list')}>
+      <BpkSectionHeader
+        title={title}
+        description={description}
+        button={button}
+      />
+
+      <div className={getClassName('bpk-card-list--card-list')}>
+        <BpkBreakpoint query={BREAKPOINTS.MOBILE}>
+          {(isActive) => {
+            if (isActive) {
+              if (layoutMobile === 'rail') {
+                return <BpkCardListRail>{cardList}</BpkCardListRail>;
+              }
+              const { accessory } = props;
+
+              let modeProps = {};
+              if (accessory === BpkAccessoryTypes.Expand) {
+                modeProps = {
+                  accessory,
+                  expandText: props.expandText,
+                  onButtonClick: props.onButtonClick,
+                };
+              } else if (accessory === BpkAccessoryTypes.Button) {
+                modeProps = {
+                  accessory,
+                  buttonText: props.buttonText,
+                  onButtonClick: props.onButtonClick,
+                };
+              }
+              return (
+                <BpkCardListStack
+                  showContent={showContent}
+                  hideContent={hideContent}
+                  collapsed={collapsed}
+                  setCollapsed={setCollapsed}
+                  {...modeProps}
+                >
+                  {visibleCards}
+                </BpkCardListStack>
+              );
+            }
+
+            if (layoutDesktop === 'row') {
+              const { accessory } = props;
+              return (
+                <BpkCardListRow
+                  accessory={button ? undefined : accessory}
+                  numberOfCardsToShow={initiallyShownCards}
+                  ariaLabelNext={props.ariaLabelNext}
+                  ariaLabelPrev={props.ariaLabelPrev}
+                  ariaLabelIndicator={props.ariaLabelIndicator}
+                >
+                  {cardList}
+                </BpkCardListRow>
+              );
+            }
+
+            const { accessory } = props;
```

Comment:
This seems a duplicate of line 87 as its stillin the same context, can we just move this up to the top outside, extract what we need and then pass it down?

As there is a lot of duplication aroun d here

Reply:
L87 is for when its in mobile and this line is for when its for desktop. I don't think its possible to extract accessory outside of the `BpkBreakpoint` then use it inside the js block for `BpkBreakpoint`